### PR TITLE
sharness/dependencies: update ipfs-update

### PR DESCRIPTION
**Pull request by chriscool**
This should fix issue #24 ('ipfs-update versions' cannot resolve name).

### CODE REVIEW

**Comment by chriscool** [comment]
There is now:

```
expecting success: 
	"$LOCAL_IPFS_UPDATE" versions >actual ||
	test_fsh cat actual

> cat actual
ERROR: fetching resource: 400 Bad Request
ERROR: Failed to query versions:  400 Bad Request: Path Resolve error: could not resolve name.

not ok 2 - 'ipfs-update versions' works
```

**Comment by chriscool** [comment]
By the way maybe ipfs-update should output error messages on stderr rather than stdout.

**Comment by whyrusleeping** [comment]
@chriscool your wish is my command: https://github.com/ipfs/ipfs-update/commit/1ca811b6c140fde07be83dcda2ea4365dc04e35b

**Comment by chriscool** [comment]
@whyrusleeping thanks, I updated the ipfs-update dependency again, but this does not fix the problem. There is still:

```
expecting success: 
	"$LOCAL_IPFS_UPDATE" versions >actual ||
	test_fsh cat actual

ERROR: Failed to query versions:  400 Bad Request: Path Resolve error: could not resolve name.
> cat actual
ERROR: fetching resource: 400 Bad Request

not ok 2 - 'ipfs-update versions' works
```

It is strange because it works on my machine now. It must be something on the Travis environment, but I don't see what.

**Comment by whyrusleeping** [comment]
thats really strange... travis' environment has always been screwy... maybe its caching the old iptb binary or something?

**Comment by whyrusleeping** [comment]
i restarted it, so hopefully we can get it happy again

**Comment by chriscool** [comment]
Yeah, we will see. Thanks anyway!

**Comment by chriscool** [comment]
Now the error is:

```
expecting success: 
	"$LOCAL_IPFS_UPDATE" versions >actual ||
	test_fsh cat actual

ERROR: Failed to query versions:  503 Service Unavailable: 
> cat actual
ERROR: fetching resource: 503 Service Unavailable

not ok 2 - 'ipfs-update versions' works
```

**Comment by chriscool** [comment]
@whyrusleeping could you restart the tests again or take another look at what could be wrong, as it is blocking everything on this repo.

**Comment by whyrusleeping** [comment]
@chriscool this is bizarre... the only way i can reproduce the error here is if i install the older version of ipfs-update and use it instead of the one vendored here. Its like travis has some weird caching going on, or it has bad dns servers? Does it fail locally for you at all?

**Comment by lgierth** [comment]
While I'm getting ahold of better nginx-level monitoring (very high on my list) -- it'd be tremendously helpful if all our little tools identified themselves with a User-Agent header if they're not doing so already. That makes debugging much easier as we can simply grep the nginx logs for e.g. "ipfs-update".

**Comment by chriscool** [comment]
@whyrusleeping no it hasn't failed locally at all since I updated ipfs-update.

**Comment by chriscool** [comment]
@whyrusleeping by the way maybe it is a good time to start using gx in fs-repo-migrations?

**Comment by chriscool** [comment]
@lgierth yeah I very much agree about each tool having its own User-Agent header, great idea!

**Comment by chriscool** [comment]
Still the same error...

**Comment by chriscool** [comment]
There is the following error:

```
*** installing bin/ipfs-update ***
go build  -o bin/ipfs-update ./dependencies/ipfs-update
dependencies/ipfs-update/main.go:11:2: cannot find package "github.com/ipfs/ipfs-update/config" in any of:
	/home/travis/.gimme/versions/go1.5.2.linux.amd64/src/github.com/ipfs/ipfs-update/config (from $GOROOT)
	/home/travis/gopath/src/github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/ipfs/ipfs-update/config (from $GOPATH)
	/home/travis/gopath/src/github.com/ipfs/ipfs-update/config
make: *** [bin/ipfs-update] Error 1
The command "make" exited with 2.
```

@whyrusleeping could you take a look at this?

File: sharness/dependencies/ipfs-update/revert.go [hunk 1]
```diff
@@ -10,8 +10,8 @@ import (
 	"text/tabwriter"
 	"time"
 
-	util "github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/ipfs/ipfs-update/util"
```
Comment (whyrusleeping) [inline sharness/dependencies/ipfs-update/revert.go]: we should redo this path rewrite

Comment (whyrusleeping) [inline sharness/dependencies/ipfs-update/revert.go]: (both of these)

File: sharness/dependencies/ipfs-update/install.go [hunk 2]
```diff
@@ -11,9 +10,9 @@ import (
 	"runtime"
 	"strings"
 
-	test "github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/ipfs/ipfs-update/test-dist"
```
Comment (whyrusleeping) [inline sharness/dependencies/ipfs-update/install.go]: should probably rewrite these too

File: sharness/dependencies/ipfs-update/test-dist/testnew.go [hunk 3]
```diff
@@ -167,8 +167,13 @@ func waitForApi(ipfspath string) error {
 }
 
 func TestBinary(bin, version string) error {
```
Comment (whyrusleeping) [inline sharness/dependencies/ipfs-update/test-dist/testnew.go]: wow, dealing with godeps is annoying... this file needs to be moved into `Godeps/_workspace/src/github.com/ipfs/ipfs-update/test-dist`

File: sharness/dependencies/ipfs-update/util/utils.go [hunk 4]
```diff
@@ -12,13 +12,14 @@ import (
 	"strings"
 
 	api "github.com/ipfs/go-ipfs-api"
+	config "github.com/ipfs/ipfs-update/config"
```
Comment (whyrusleeping) [inline sharness/dependencies/ipfs-update/util/utils.go]: this file should also be moved into a godeps path just like the one above, and then this path will have to be rewritten to the godeps one...

**Comment by whyrusleeping** [comment]
This is just a bunch of obnoxious godeps stuff... If you want to push your code to a branch on this repo (you have collab) I can fix it up for you pretty quickly (i've unfortunately done this a lot).

I would normally recommend we just use gx here, but theres so much code in godeps in different weird ways that I don't want to deal with all of it.

**Comment by chriscool** [comment]
@whyrusleeping ok, I pushed the code into the "update-ipfs-update" branch. Thanks!

**Comment by chriscool** [comment]
This branch: https://github.com/ipfs/fs-repo-migrations/tree/update-ipfs-update

**Comment by whyrusleeping** [comment]
@chriscool thats really weird... i pushed to that branch. and github says its this PR, but my commit isnt on this PR.

**Comment by chriscool** [comment]
Closing in favor of PR #27 that has @whyrusleeping's commit.

**Pull request closed without merging**